### PR TITLE
[KYUUBI #4696] Upgrade scalafmt from 3.7.1 to 3.7.3

### LINKED ISSUE #4696: :arrow_up: Upgrade scalafmt from 3.7.1 to 3.7.3
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Why do we need to upgrade this artifact?

Bugfixes

### Artifact Name

scalafmt

### Target Version

3.7.3

### Notable Changes

https://github.com/scalameta/scalafmt/releases/tag/v3.7.3

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [X] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by huapan123456**
### _Why are the changes needed?_

Close #4696

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -245,7 +245,7 @@
         <spotless.java.googlejavaformat.version>1.7</spotless.java.googlejavaformat.version>
         <spotless.python.includes></spotless.python.includes>
         <spotless.python.black.version>22.3.0</spotless.python.black.version>
-        <spotless.scala.scalafmt.version>3.7.1</spotless.scala.scalafmt.version>
+        <spotless.scala.scalafmt.version>3.7.3</spotless.scala.scalafmt.version>
```
Comment (pan3793) [inline pom.xml:248]: `.scalafmt.conf` should be updated as well, and let's add a comment to mention it here.

File: pom.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
         <artifactId>apache</artifactId>
         <version>29</version>
     </parent>
-
+R
```
Comment (pan3793) [inline pom.xml:27]: unexpected change

File: pom.xml (hunk 1 above)
Comment (huapan123456) [reply pom.xml:248]: done

File: pom.xml (hunk 2 above)
Comment (huapan123456) [reply pom.xml]: done

File: pom.xml (hunk 1 above)
Comment (yaooqinn) [inline pom.xml:248]:
```suggestion
        <!-- Please also update .scalafmt.conf when you change it here -->
        <spotless.scala.scalafmt.version>3.7.3</spotless.scala.scalafmt.version>
```

Comment (huapan123456) [reply pom.xml:249]: done

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**